File: app/src/main/java/com/example/pam_app/view/AddBucketView.java
package com.example.pam_app.view;

import com.example.pam_app.model.Bucket;

public interface AddBucketView {

    void onErrorSavingBucket();
    void onSuccessSavingBucket(final Bucket bucket);
    void goToLoadImage();
    void showTitleError(final int error, final Integer parameter);
    void showTargetError(final int error, final Integer parameter);
    void showDateError(final int error);
    void showBucketTypeError(final int error);
}

File: app/src/main/java/com/example/pam_app/view/AddBucketView.kt
package com.example.pam_app.view

import com.example.pam_app.model.Bucket

interface AddBucketView {
    fun onErrorSavingBucket()
    fun onErrorExistingBucketName()
    fun onSuccessSavingBucket(bucket: Bucket?)
    fun showTitleError(error: Int, parameter: Int?)
    fun showTargetError(error: Int, parameter: Int?)
    fun showDateError(error: Int)
    fun showBucketTypeError(error: Int)
    fun requestStoragePermission()
    fun changeDatePickerState(state: Boolean)
    fun setDefaultValues(bucket: Bucket?)
}